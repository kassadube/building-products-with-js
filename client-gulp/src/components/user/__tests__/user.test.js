import configureMockStore from 'redux-mock-store';
import UserWarraper, {User} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: '2017-05-07T10:22:37.364Z',
};


test('# User Warraper', () => {
    const store = mockStore({auth: user});
  const component = (<UserWarraper user={user} store={store}/>);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});


test('# User', () => {
  const component = (<User user={user} />); 
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});
