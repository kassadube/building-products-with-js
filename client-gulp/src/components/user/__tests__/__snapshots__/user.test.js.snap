// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# User 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    User: 
    test
     
  </div>
  <div
    className="panel-body"
  >
    Registration Date 
    2017-05-07T10:22:37.364Z
  </div>
</div>
`;

exports[`# User Warraper 1`] = `
<User
  edit={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  updateUser={[Function]}
  user={
    Object {
      "id": "0",
      "login": "test",
      "registrationDate": "2017-05-07T10:22:37.364Z",
    }
  }
/>
`;
